File: DependencyInjection/ConformingContainerAntipattern/_3_ConformingContainer/Outbound/OutboundMessage.cs
using ConformingContainerAntipattern._3_ConformingContainer.Interfaces;

namespace ConformingContainerAntipattern._3_ConformingContainer.Outbound
{
  public interface IOutboundMessage : DataDestination
  {
    void SendVia(IOutputSocket outputOutputSocket);
  }

  class OutboundMessage : IOutboundMessage
  {
    private readonly IMarshalling _marshalling;
    private string _content = string.Empty;

    public OutboundMessage()
    {
      _marshalling = ApplicationRoot.Context.Resolve<IMarshalling>();
    }

    public void SendVia(IOutputSocket outputOutputSocket)
    {
      var marshalledContent = _marshalling.Of(_content);
      outputOutputSocket.Open();
      outputOutputSocket.Send(marshalledContent);
      outputOutputSocket.Close();
    }

    public void Add(string s)
    {
      _content += s;
    }
  }
}

File: Java/dependency-injection/_06_conforming-container-antipattern/src/main/java/com/github/grzesiek_galezowski/ConformingContainerAntipattern/Outbound/OutboundMessage.java
package com.github.grzesiek_galezowski.ConformingContainerAntipattern.Outbound;

import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Interfaces.DataDestination;

public interface OutboundMessage extends DataDestination {
  void sendVia(OutputSocket outputOutputSocket);
}
